term/src/main/jni/termExec.cpp

```diff
@@ -71,8 +70,8 @@ static void android_os_Exec_setPtyUTF8Mode(JNIEnv *env, jobject clazz, jint fd,
         tios.c_iflag &= ~IUTF8;
     }
 
-    if (my_tcsetattr(fd, TCSANOW, &tios) < 0)
```

Comment:
This use of `TCSANOW` is where issue seems to lie. Compare https://android.googlesource.com/platform/bionic/+/master/libc/bionic/termios.cpp and https://android.googlesource.com/platform/bionic/+/android-4.2_r1/libc/include/termios.h (hint: the former does not redefine `TCSANOW` at all).